common/src/main/java/com/wynntils/core/notifications/NotificationManager.java

```diff
@@ -29,6 +43,25 @@ public static MessageContainer queueMessage(TextRenderTask message) {
 
         WynntilsMod.info("Message Queued: " + message);
         MessageContainer msgContainer = new MessageContainer(message);
+        String messageText = message.getText();
+
+        for (Pair<String, MessageContainer> cachedMessagePair : cachedMessageSet) {
+            String checkableMessage = cachedMessagePair.a();
+            if (messageText.equals(checkableMessage)) {
+                MessageContainer matchedContainer = cachedMessagePair.b();
+                matchedContainer.incrementMessageCount();
+
+                // Overlay is not enabled, send in chat
+                if (!GameNotificationOverlayFeature.INSTANCE.isEnabled()) {
+                    sendOrEditNotification(msgContainer);
+                }
+
+                WynntilsMod.postEvent(new NotificationEvent.Edit(matchedContainer));
```

Comment:
Can you extract line 55-59 into a separate `doSendMessage` (or something like that), and also replace the code at line 66-71 with a call to the same function?

I also really don't like that we check a feature from a manager, but I have no good suggesion at hand on how to solve this otherwise. But we should at least not duplicate the code.

Other than this, the PR is starting to look really, really good!